### DISCUSSION TITLE
TLS decryption in Wireshark - server responses remain encrypted

### DISCUSSION THREAD

**Discussion by skomialek** [mitmproxy/mitmproxy#6160] [Q&A] [answered]
Hi,

I'm trying to use `mitmproxy` as a tool for exporting `SSLKEYLOGFILE` to decrypt traffic captured with Wireshark. I managed to set up `mitmproxy` in transparent mode and route the traffic through it, it all looks good, and everything is functional. 

At the same time, I capture the traffic in Wireshark and provide the exported `SSLKEYLOGFILE` to see the data, and to my surprise it decrypts only half of the captured traffic - the client requests. Server responses remain decrypted. 

My setup

`CLIENT (192.168.43.15)  <->  PROXY (192.168.43.13) <-> SERVER (192.168.43.10)`

The client makes a simple HTTPS GET request to the service running on the server port 9999. The Wireshark capture happens on the `PROXY` machine (and this is where `mitmproxy` runs and intercepts the traffic in transparent mode).

As mentioned before, I do get client requests decrypted but not server responses.

<img width="962" alt="image" src="https://github.com/mitmproxy/mitmproxy/assets/12737549/1014ea53-a7e0-4e5d-935e-368eda93733d">

I was wondering if something is missing in the exported `SSLKEYLOGFILE` preventing Wireshark from decrypting the traffic?

**Comment by mhils** [comment, marked as answer]
Wireshark somehow fails to display that properly, but the decrypted response is there as well. Try right click -> follow -> TLS stream. :-)

**Reply by skomialek** [reply]
Thank you, that's probably Wireshark bug :)
